lib/services/access_request_cache_test.go

```diff
@@ -60,6 +66,92 @@ func newAccessRequestPack(t *testing.T) (accessRequestServices, *services.Access
 	return svcs, cache
 }
 
+func TestAccessRequesstCacheResets(t *testing.T) {
+	const (
+		requestCount = 1000
+		workers      = 100
+		resets       = 3
+	)
+
+	t.Parallel()
+
+	svcs, cache := newAccessRequestPack(t)
+	defer cache.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for i := 0; i < requestCount; i++ {
+		r, err := types.NewAccessRequest(uuid.New().String(), "alice@example.com", "some-role")
+		require.NoError(t, err)
+
+		_, err = svcs.CreateAccessRequestV2(ctx, r)
+		require.NoError(t, err)
+	}
+
+	timeout := time.After(time.Second * 30)
+
+	for {
+		rsp, err := cache.ListAccessRequests(ctx, &proto.ListAccessRequestsRequest{
+			Limit: requestCount,
+		})
+		require.NoError(t, err)
+		if len(rsp.AccessRequests) == requestCount {
+			break
+		}
+
+		select {
+		case <-timeout:
+			require.FailNow(t, "timeout waiting for access request cache to populate")
+		case <-time.After(time.Millisecond * 200):
+		}
+	}
+
+	doneC := make(chan struct{})
+	var eg errgroup.Group
+
+	for i := 0; i < workers; i++ {
+		eg.Go(func() error {
+			for {
+				select {
+				case <-doneC:
+					return nil
+				case <-time.After(time.Millisecond * 10):
```

Comment:
10 milliseconds doesn't feel like enough time for anything to happen in CI.

Reply:
Backed it off a bit and added an additional check in the write-side loop to verify that a minimum number of concurrent reads happen for every tick.  Ultimately though we want this part to be pretty high frequency though, since part of the purpose of this test case is to catch problems with background resets not being seamless.